sdk/azcore/policy_retry.go

```diff
@@ -19,27 +20,31 @@ const (
 )
 
 // RetryOptions configures the retry policy's behavior.
+// All zero-value fields will be initialized with their default values.
 type RetryOptions struct {
 	// MaxRetries specifies the maximum number of attempts a failed operation will be retried
-	// before producing an error.  A value of zero means one try and no retries.
+	// before producing an error.
+	// The default value is three.  A value less than zero means one try and no retries.
 	MaxRetries int32
```

Comment:
What if we called this MaxTries: 0=default, 1=1 try, 2=2 tries, 3=3 tries, etc.
This way there is no -1 -- it just bugs me.

Reply:
We can do that, and that will address the most common cases.  However, we still have to do _something_ about a negative value since the type allows it even though it semantically makes no sense.  As an aside, we had a bug a few years ago around this exact thing in track 1; some incorrect math was leading to a negative try value, so the request was never even sent leading to a very hard-to-diagnose bug.  It would be preferable to prevent the possibility of such bugs from even occurring (isn't this part of the value-add of an SDK?).

Reply:
If we change it to `MaxTries` then we must return an error from `retryPolicy.Do()` if `MaxTries` is negative.

Reply:
This is the argument for having the XxxClient ctors return an error if any option does not pass validation.

Reply:
We could, however I think it's a high tax to pay (making the ctors not no-fail) when we can design the API in a way to avoid this.

0 == default
> 0 == custom value
< 0 == none

I think this is very clear and assigns semantic meaning to the complete range of an `int`.  There is nothing to validate, all values are valid.